### DISCUSSION TITLE
[Composer] How do I organize a dashboard on different project

### DISCUSSION THREAD

**Discussion by Neo1975** [mia-platform/community#363] [Q&A] [answered]
### Which section?

Backoffice

### Which is the version?

12

### Description

Hi,
I have a tenant with many different project. I would like to create a unique dashboard where users can find out information from different project. I'm making up a Composer page with a table, just to start and test solution, but Composer can use only one crud-service to retrive information. The problem that is that I would like to setup a independent project (Internal Dashboard) with many different pages with data from other project. How do you suggest me to organize it? Should I have to create a Composer project for each "sub"-project and reference as IFRAME in the Internal DashBoard? Are there ìother solutions?

Thanks.

**Comment by epessina** [comment, marked as answer]
Hi @Neo1975, thanks for reaching out.

You can connect to multiple CRUD services from the same Composer page instantiating multiple [crud clients](https://docs.mia-platform.eu/docs/microfrontend-composer/back-kit/components/crud_client) with different [base path](https://docs.mia-platform.eu/docs/microfrontend-composer/back-kit/components/crud_client#properties--attributes) properties and routing the calls to the correct service through the Gateway of your project.
However, I would not recommend this approach, since you will likely incur authentication and [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) errors.

While the solution you have proposed with iFrames is perfectly doable, I would suggest creating an ad-hoc BFF for the Composer in the Dashboard project (so that no CORS errors may arise). This BFF should contact the CRUD services (M2M Authorization) and expose the data to the Composer from a single source. If you build the BFF in a way that exposes the same interface as a CRUD service you can still use a `bk-crud-client` for communication. On the other hand, if you decide to expose your own interface, you will have to built a custom "client" web component.
